### ISSUE TITLE
Segment fault on linux

### ISSUE THREAD

**Issue by buddyjojo** [arnavbhatt288/ros-ramdisk-creator-gui#4]
```
execve("./src/ramdisk-creator", ["./src/ramdisk-creator"], 0x7ffcfa6672f0 /* 18 vars */) = 0
brk(NULL)                               = 0x55f8cb64a000
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f3b20ae1000
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
newfstatat(3, "", {st_mode=S_IFREG|0644, st_size=231002, ...}, AT_EMPTY_PATH) = 0
mmap(NULL, 231002, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f3b20aa8000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libX11.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\0\0\0\0\0\0\0"..., 832) = 832
newfstatat(3, "", {st_mode=S_IFREG|0644, st_size=1318408, ...}, AT_EMPTY_PATH) = 0
mmap(NULL, 1318216, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f3b20966000
mmap(0x7f3b20982000, 569344, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1c000) = 0x7f3b20982000
mmap(0x7f3b20a0d000, 610304, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xa7000) = 0x7f3b20a0d000
mmap(0x7f3b20aa2000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x13c000) = 0x7f3b20aa2000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libblkid.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\0\0\0\0\0\0\0"..., 832) = 832
newfstatat(3, "", {st_mode=S_IFREG|0644, st_size=355328, ...}, AT_EMPTY_PATH) = 0
mmap(NULL, 354104, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f3b2090f000
mmap(0x7f3b20918000, 221184, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x9000) = 0x7f3b20918000
mmap(0x7f3b2094e000, 69632, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x3f000) = 0x7f3b2094e000
mmap(0x7f3b2095f000, 28672, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x50000) = 0x7f3b2095f000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0Ps\2\0\0\0\0\0"..., 832) = 832
pread64(3, "\6\0\0\0\4\0\0\0@\0\0\0\0\0\0\0@\0\0\0\0\0\0\0@\0\0\0\0\0\0\0"..., 784, 64) = 784
newfstatat(3, "", {st_mode=S_IFREG|0755, st_size=1922136, ...}, AT_EMPTY_PATH) = 0
pread64(3, "\6\0\0\0\4\0\0\0@\0\0\0\0\0\0\0@\0\0\0\0\0\0\0@\0\0\0\0\0\0\0"..., 784, 64) = 784
mmap(NULL, 1970000, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f3b2072e000
mmap(0x7f3b20754000, 1396736, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x26000) = 0x7f3b20754000
mmap(0x7f3b208a9000, 339968, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x17b000) = 0x7f3b208a9000
mmap(0x7f3b208fc000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1ce000) = 0x7f3b208fc000
mmap(0x7f3b20902000, 53072, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f3b20902000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libxcb.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\0\0\0\0\0\0\0"..., 832) = 832
newfstatat(3, "", {st_mode=S_IFREG|0644, st_size=166824, ...}, AT_EMPTY_PATH) = 0
mmap(NULL, 169096, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f3b20704000
mmap(0x7f3b20710000, 77824, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xc000) = 0x7f3b20710000
mmap(0x7f3b20723000, 36864, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1f000) = 0x7f3b20723000
mmap(0x7f3b2072c000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x27000) = 0x7f3b2072c000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libXau.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\22\0\0\0\0\0\0"..., 832) = 832
newfstatat(3, "", {st_mode=S_IFREG|0644, st_size=14496, ...}, AT_EMPTY_PATH) = 0
mmap(NULL, 16672, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f3b206ff000
mmap(0x7f3b20700000, 4096, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1000) = 0x7f3b20700000
mmap(0x7f3b20701000, 4096, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0x7f3b20701000
mmap(0x7f3b20702000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0x7f3b20702000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libXdmcp.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0@\23\0\0\0\0\0\0"..., 832) = 832
newfstatat(3, "", {st_mode=S_IFREG|0644, st_size=22728, ...}, AT_EMPTY_PATH) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f3b206fd000
mmap(NULL, 4215024, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f3b202f7000
mmap(0x7f3b20400000, 2117872, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0) = 0x7f3b20400000
munmap(0x7f3b202f7000, 1085440)         = 0
munmap(0x7f3b20606000, 1007856)         = 0
mprotect(0x7f3b20405000, 2093056, PROT_NONE) = 0
mmap(0x7f3b20604000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x4000) = 0x7f3b20604000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libbsd.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\0\0\0\0\0\0\0"..., 832) = 832
newfstatat(3, "", {st_mode=S_IFREG|0644, st_size=84840, ...}, AT_EMPTY_PATH) = 0
mmap(NULL, 86224, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f3b206e7000
mmap(0x7f3b206eb000, 49152, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x4000) = 0x7f3b206eb000
mmap(0x7f3b206f7000, 12288, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x10000) = 0x7f3b206f7000
mmap(0x7f3b206fa000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x13000) = 0x7f3b206fa000
mmap(0x7f3b206fc000, 208, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f3b206fc000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libmd.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\0\0\0\0\0\0\0"..., 832) = 832
newfstatat(3, "", {st_mode=S_IFREG|0644, st_size=47312, ...}, AT_EMPTY_PATH) = 0
mmap(NULL, 49384, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f3b206da000
mmap(0x7f3b206dc000, 28672, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0x7f3b206dc000
mmap(0x7f3b206e3000, 8192, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x9000) = 0x7f3b206e3000
mmap(0x7f3b206e5000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xa000) = 0x7f3b206e5000
close(3)                                = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f3b206d8000
arch_prctl(ARCH_SET_FS, 0x7f3b206d8b80) = 0
set_tid_address(0x7f3b206d8e50)         = 206526
set_robust_list(0x7f3b206d8e60, 24)     = 0
rseq(0x7f3b206d94a0, 0x20, 0, 0x53053053) = 0
mprotect(0x7f3b208fc000, 16384, PROT_READ) = 0
mprotect(0x7f3b206e5000, 4096, PROT_READ) = 0
mprotect(0x7f3b206fa000, 4096, PROT_READ) = 0
mprotect(0x7f3b20604000, 4096, PROT_READ) = 0
mprotect(0x7f3b20702000, 4096, PROT_READ) = 0
mprotect(0x7f3b2072c000, 4096, PROT_READ) = 0
mprotect(0x7f3b2095f000, 24576, PROT_READ) = 0
mprotect(0x7f3b20aa2000, 4096, PROT_READ) = 0
mprotect(0x55f8c9a7b000, 4096, PROT_READ) = 0
mprotect(0x7f3b20b13000, 8192, PROT_READ) = 0
prlimit64(0, RLIMIT_STACK, NULL, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
munmap(0x7f3b20aa8000, 231002)          = 0
getrandom("\xad\x07\xae\x30\xd6\x6c\x91\x60", 8, GRND_NONBLOCK) = 8
brk(NULL)                               = 0x55f8cb64a000
brk(0x55f8cb66b000)                     = 0x55f8cb66b000
openat(AT_FDCWD, "/proc/partitions", O_RDONLY) = 3
newfstatat(3, "", {st_mode=S_IFREG|0444, st_size=0, ...}, AT_EMPTY_PATH) = 0
read(3, "major minor  #blocks  name\n\n 259"..., 1024) = 258
read(3, "", 1024)                       = 0
lseek(3, 0, SEEK_SET)                   = 0
read(3, "major minor  #blocks  name\n\n 259"..., 1024) = 258
read(3, "", 1024)                       = 0
close(3)                                = 0
openat(AT_FDCWD, "/dev/nvme1n1/dev/nvme1n1/dev/nvme0n1/dev/nvme0n1/dev/nvme0n1/dev/sde1", O_RDONLY|O_NONBLOCK|O_CLOEXEC) = -1 ENOTDIR (Not a directory)
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x38} ---
+++ killed by SIGSEGV (core dumped) +++
Segmentation fault
```

**Comment by abt2804** [comment]
@buddyjojo Did you run the program with sudo or not?

**Comment by buddyjojo** [comment]
yes

**Comment by abt2804** [comment]
> yes

Odd, so it immediately crashes right after running it?

**Comment by buddyjojo** [comment]
yep, it just says segment fault and nothing else (output above was with strace)

**Comment by abt2804** [comment]
> yep, it just says segment fault and nothing else (output above was with strace)

Can you show me the output of /proc/partitions`?

**Comment by buddyjojo** [comment]
```
major minor  #blocks  name

 259        0  976762584 nvme1n1
 259        5  976760832 nvme1n1p1
 259        2  976762584 nvme0n1
 259        3     524288 nvme0n1p1
 259        4  975236096 nvme0n1p2
   8       64    7839744 sde
   8       65    7838720 sde1
```

**Comment by abt2804** [comment]
I probably have to find a better way to gather all partitions mounted. The current implementation is not robust.

**Comment by buddyjojo** [comment]
Ah ok

**Comment by abt2804** [comment]
Issue resolved at [cf9c4d4](https://github.com/arnavbhatt288/ros-ramdisk-creator-gui/commit/cf9c4d46ae430ce3750314d497a82de448cfd4c5).
